compiler/go.vim

```diff
@@ -5,6 +5,7 @@ let g:current_compiler = 'go'
 let s:save_cpo = &cpoptions
 set cpoptions-=C
 
+" CompilerSet makeprg=go
```

Comment:
Will this also take the logic to add the build flags etc. below? Or will you have to do that manually?

Reply:
Dispatch is (to me) an odd plugin. It does what I want and has for years,but I only just figured out how it works (because it wasn't working with gopher). When you use the 'Dispatch' command you can give it one argument, the name of a program, (and possibly arguments to that program). It then searches through *all* compiler files in &rtp and looks for a string like "CompilerSet makeprg=<thing>".  If it finds a match it runs :compiler <compiler name>, sets &makeprg to the given <program>, and runs :make in the background.  The point is to run a program with the &efm of a particular compiler.
e.g.
If I run 
:Dispatch go test
Dispatch will search all compiler files until it comes to a file that has a line that looks like
CompilerSet makeprg=go\ test
In my case that file would be compilers/gotest.vim. So Dispatch will run 
:compiler gotest
It will then set &makeprg to "go test" and will run :make in the background (using one of several techniques). 

You can also do other things with dispatch, such as:
:Dispatch -compiler gotest someOtherProgram
which will run
:compiler gotest
and then will set &makeprg "someOtherProgram", and run :make in the background.

I don't use :Dispatch a whole lot, but when I do it is convenient.  I mostly use its :Make command which simply runs :make asynchronously. So I've defined several commands that look something like "command GoBuild compiler gotest | Make" (they are a bit more complicated but that's the idea). 

The comments in this PR using CompilerSet are useful when I (or anyone else) needs to use :Dispatch in its simplest form.
I assume dispatch is looking for CompilerSet lines since that seems to be the recommended way of setting &makeprg and &efm these days.

Reply:
Alright; thanks. I added a line of documentation on it and merged ot